### ISSUE TITLE
Jco transpile: implementation of wasi-nn for Web

### ISSUE THREAD

**Issue by danbev** [bytecodealliance/jco#407]
This issue is intended for tracking adding a shim for wasi-nn.

Tasks:
- [ ] ...
- [ ] ...

**Comment by danbev** [comment]
Hope it is alright that I created this issue without any actual tasks. I'm not sure what needs to be done or how these shims are created but hopefully others might be able to add some tasks for this.

**Comment by guybedford** [comment]
This would be great to see! The goal for Jco is to support all WASI staged proposals automatically so they can work out of the box. Some details on how the integration can be designed via automatic map configuration is provided in https://bytecodealliance.github.io/jco/transpiling.html#wasi-proposals. By posting this issue you have now volunteered you realise :P

**Comment by vados-cosmonic** [comment]
Hey @danbev were you able to find any more time to work on this?

It looks like this  functionality actually doesn't need explicit support *inside* jco but I would be happy to upstream a repo (similar to `preview2-shim`, for example) that made use of `jco transpile`'s functionality for `--map`ing imports -- I don't think the actual work is unblocked now :)

**Comment by vados-cosmonic** [comment]
I don't think this effort ever quite got off the ground so I'm going to close this for now -- hopefully someone can get back to this in the future, and we can re-open it then.
